### PR TITLE
Linking anonymous account to phone number

### PR DESCRIPTION
**Pull request by creativecreatorormaybenot**
This is lacking tests (although I do not know if they are required) and testing from my side. I am currently unable to test it because of [this issue](https://github.com/flutter/flutter/issues/22568), but I figured that I might want to get some feedback already.

This pull request would resolve https://github.com/flutter/flutter/issues/22491.

### CODE REVIEW

**Comment by kdy1** [comment]
This also workarounds https://github.com/flutter/flutter/issues/22304, as I can authenticate user with email first.

**Comment by creativecreatorormaybenot** [comment]
@kdy1 Good point, this definitely has to be added to this pull request because it is unwanted behavior.

**Comment by creativecreatorormaybenot** [comment]
I can confirm that this is now fully working as expected.  
I am not sure what the point of the test functions is, but I added it anyway.

**Comment by kdy1** [comment]
@creativecreatorormaybenot  Does this pr add a feature to send a verification code?

**Comment by creativecreatorormaybenot** [comment]
@kdy1 This is what `verifyPhoneNumber` is doing already.

File: packages/firebase_auth/lib/firebase_auth.dart [hunk 1]
```diff
@@ -386,6 +386,24 @@ class FirebaseAuth {
     return currentUser;
   }
 
+  /// Links phone number with current user and returns [Future<FirebaseUser>], which will contain the phone number information.
```
Comment (kf6gpe) [inline packages/firebase_auth/lib/firebase_auth.dart]: Nit: Break long line please.

**Comment by kf6gpe** [comment]
This looks good, except for one teensy formatting nit, if you don't mind. Thanks for the contribution, and thanks especially for including a test with it! If you don't mind updating for the nit and resolving the branch conflict, we'll merge this after. Otherwise, we'll get it picked up eventually.

**Comment by collinjackson** [comment]
I can take a look at this if conflicts are resolved

**Comment by creativecreatorormaybenot** [comment]
This is now possible using `linkWithCredential` and `PhoneAuthProvider.getCredential` [as described here](https://stackoverflow.com/a/54393307/6509751).

**Comment by awazgyawali** [comment]
Although this is possible using PhoneAuthProvider.getCredential , andorid automatically detectes sms code and signs in the user with newer account instead of linking with the current one.

**Comment by collinjackson** [comment]
You're right @awazgyawali, see #22304

**Comment by awazgyawali** [comment]
Thus I have added a PR to fix this, can you look into it? Its a very simple one. https://github.com/flutter/plugins/pull/1568

**Pull request closed without merging**